ci: add abidiff workflow

Tachi107 opened the pull request:
This CI workflow checks ABI compatibility between the pushed commit and the previous, helping preventing accidental ABI breaks.

Helps with https://github.com/yhirose/cpp-httplib/issues/2043

Cc: @mgorny

Tachi107 commented:
Note: I'm not sure this works correctly on pull requests, since they may contain more than one commit.

yhirose commented on .github/workflows/abidiff.yaml:
```diff
@@ -0,0 +1,62 @@
+
+    - name: Checkout previous
+      working-directory: previous
+      run: git checkout HEAD^
```
Could you try `git checkout $(git describe --tags --abbrev=0 master)`?
`$(git describe --tags --abbrev=0 master)` gives us the commit hash for the latest version tag.

Tachi107 replied:
You mean diffing the current commit with the previous release? Brilliant! Didn't think about it :)

yhirose commented:
> Note: I'm not sure this works correctly on pull requests, since they may contain more than one commit.

Thanks a lot for taking care of it! I made a comment about how to get the commit hash for the latest version tag. Could you try with it to see how it goes?

Tachi107 commented:
> Thanks a lot for taking care of it! I made a comment about how to get the commit hash for the latest version tag. Could you try with it to see how it goes?

Done! The job now fails because it has detected breaking changes since the last tagged release, and abidiff returns a status code of `12`, which means `ABIDIFF_ABI_CHANGE` and `ABIDIFF_ABI_INCOMPATIBLE_CHANGE` according to its [manpage](https://sourceware.org/libabigail/manual/abidiff.html).

yhirose commented:
Thanks for the great work!
